### DISCUSSION TITLE
Best method for knowing that the character is "In Action"? Rolling, jumping, mantling, etc.

### DISCUSSION THREAD

**Discussion by xkamronx** [Sixze/ALS-Refactored#283] [Q&A]
For general gameplay logic, what would you suggest for knowing that the character shouldn't be able to do another action, like melee attacking or reloading?

Right now, I'm doing a check for the locomotion tags before starting stuff, but it's not exactly solid logic. Having something that's a set in stone, "I'm busy, can't do that right now" would be super useful for basically any game I'd think.

**Comment by assmikDuDuDuJ** [comment]
What's wrong with tags? I'm a beginner, but I think tags are pretty useful, especially for those who are bad at C++.
You can check them manually before performing an action, or create a listener function for automation and bind the desired tags to the dispatches. or call events.
....
this is my opinion as a newbie.

**Reply by xkamronx** [reply]
Checking the tags is what I'm doing right now, but it's multiple checks instead of a single, "character is busy" type check, if that makes sense.

**Reply by assmikDuDuDuJ** [reply]
"multiple checks instead of a single"  do you add "has tag" branch\or\and every time you check a tag, or did you make a function that checks the selected tags by tag container\array?
If not, make a function...and you'll have "One Check".
or how do you imagine this? how do you think the "character is busy" function will know that the character is busy? it also needs to get this data somehow, for example by doing same "multiple checks" of the same tags, and give  output "character is busy".

**Comment by oivio** [comment]
Tags are one of the best parts of this plugin ;) 
I would only change thing about them is to keeping tags inside DefaultGameplayTags.ini instead of hard coded at least for overlay parts.
